### DISCUSSION TITLE
[BUG] Group License: pyarmor reg in container fails: could not get docker host machine id

### DISCUSSION THREAD

**Discussion by BenjaminYde** [dashingsoft/pyarmor#1395] [Q&A]
Pyarmor version: 8.2.9

**Device registration**

The following command fails inside the docker container:
```bash
pyarmor reg .pyarmor/pyarmor-device-regfile-xxxx.1.zip
```

Output:
```
ERROR    could not get docker host machine id
Traceback (most recent call last):
  File "/home/xxx/.local/lib/python3.10/site-packages/pyarmor/cli/register.py", line 183, in _get_docker_hostname
    s.connect((host, port))
ConnectionRefusedError: [Errno 111] Connection refused

```

**Docker info:**
- docker version: 24.0.3
- API version: 1.43

**Container info:**
- x86_64
- ubuntu 22.04
- python 3.10.6
- has the extra host: `--add-host=host.docker.internal:host-gateway`
- .pyarmor folder from host is copied into container (`~/.pyarmor`)
- pyarmor is installed using `pip install pyarmor==8.2.9`

It works fine when running on the docker host (not inside a container) but fails inside the container

**Comment by jondy** [comment]
Do you run `pyarmor-auth pyarmor-device-regfile-xxxx.1.zip` in the host machine?

Refer to
https://pyarmor.readthedocs.io/en/stable/how-to/register.html#run-unlimited-dockers-in-offline-device
